### ISSUE TITLE
Attachment names are not always parsed properly from MimeMessage

### ISSUE THREAD

**Issue by jkxyx205** [bbottema/simple-java-mail#175]
**Repo maintainer's note:** Summarizing, the issue is that resource names that are wrapped inside <..> are not read back properly, resulting in double extensions.

---

Hi Benny Bottema,

1. When I send  email with attachment `a.txt`
```java
EmailBuilder.startingBlank()
 .withAttachment("a.txt", new ByteArrayDataSource("Black Tie Optional", "text/plain"))
```
2. but get attachent name `a.txt.txt` which is not expect.
```java
Email email = EmailConverter.mimeMessageToEmail(message);
if (email.getAttachments().size() > 0) {
                System.out.println(email.getAttachments().get(0).getName());
            }
```

 ---

When I trace source code,I found 

* send email . "Content-ID" default <fileName> <a.txt>
```java
attachmentPart.setHeader("Content-ID", format("<%s>", resourceName));
```
* get attachment name
```java
        @Nonnull
	private static String parseResourceName(@Nullable final String contentID, @Nonnull final String fileName) {
		String extension = "";
		if (!valueNullOrEmpty(fileName) && fileName.contains(".")) {
			extension = fileName.substring(fileName.lastIndexOf("."), fileName.length());
		}
		if (!valueNullOrEmpty(contentID)) { 
                        // execute here
			return (contentID.endsWith(extension)) ? contentID : contentID + extension;
		} else {
			return fileName;
		}
	}
```

How can i get the right file name 'a.txt'. Look forward to your relay, thank you.

**Comment by jkxyx205** [comment]
I can get real file name by the code below:
```java
email.getAttachments().get(0).getDataSource().getName();
```
My original purpose is to get a mail from inbox,  and then use it's attachment to create new mail to send, the file name not right.

full code blow:
```java
@Test
public void forward() throws MessagingException {
    Session session = ServerConfigUtils.getSession();

    Store store = session.getStore("imap");

    store.connect(Constants.imap_host, Constants.from, Constants.password);

    Folder folder = store.getFolder("INBOX");

    folder.open(Folder.READ_WRITE);

    Message message = folder.getMessage(2);

    // get mail from inbox
    Email oldEmail = EmailConverter.mimeMessageToEmail((MimeMessage) message);

    // get use old mail content
    Email newEmail = EmailBuilder.startingBlank()
            .from("Rick", Constants.from)
            .to("Ashley", Constants.email)
            .withSubject("转发：this is subject")
            // from old mail
            .withEmbeddedImages(oldEmail.getEmbeddedImages())
            .withAttachments(oldEmail.getAttachments())
            .withHTMLText(oldEmail.getHTMLText())
            .prependTextHTML("this is my own message")
            .buildEmail();


    Mailer mailer = MailerBuilder
            .withTransportStrategy(TransportStrategy.SMTP)
            .withSMTPServer(Constants.host, 25, Constants.from, Constants.password)
            .buildMailer();


    mailer.sendMail(newEmail);

}
```

**Comment by bbottema** [comment]
This seems like a bug to me. Looking into it now.

**Comment by bbottema** [comment]
Ahh, here's what I found in my test code:

```
ByteArrayDataSource namedAttachment = new ByteArrayDataSource("Black Tie Optional", "text/plain");
namedAttachment.setName("dresscode.txt"); // normally not needed, but otherwise the equals will fail
```
If I remove that line, the test fails and this is the comparison result:
![image](https://user-images.githubusercontent.com/2115718/47603573-14d8da80-d9ee-11e8-9445-4a74fcb525fd.png)
So exactly the same as your situation. I will see what I can do to fix this properly.

**Comment by bbottema** [comment]
Fix release under 5.0.7. Please verify.

**Comment by jkxyx205** [comment]
The bug has been fixed, thank you very much for high efficient modification. **@bbottema**
